### DISCUSSION TITLE
custom properties on Connection object for Socket library

### DISCUSSION THREAD

**Discussion by Tankonyako** [reactphp/reactphp#493] [Ideas]
I think custom properties for connection, is useful for example: you have multiple modules and want transfer data between by 'data' events for example, or hold user auth.
Like i do: https://github.com/Tankonyako/reactphp-socket-with-properties

**Comment by SimonFrings** [comment]
@Tankonyako thanks for bringing this up and thank you for sharing :+1:

It's a nice idea and I think this will work for some specific use cases , but I am not sure if it will be part of ReactPHP in the nearer future. Don't get me wrong, I like this approach, but we also have to think of the bigger picture here. Not sure how needed this feature is and what it involves, particularly when  it comes to documenting, testing and recommending this.

Nonetheless, if your fork works for your use cases I think it's a win :+1:
